src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -918,6 +921,26 @@ public String getFilter() {
     public void setHeaders(Map<String, Object> requestHeadersParam) {
         properties.put(ReplicationField.REQUEST_HEADERS, requestHeadersParam);
         replicationInternal.setHeaders(requestHeadersParam);
+        // Fix - https://github.com/couchbase/couchbase-lite-java-core/issues/1617
+        storeCookiesIntoCookieJar(requestHeadersParam);
+    }
+
+    private void storeCookiesIntoCookieJar(Map<String, Object> requestHeadersParam) {
+        try {
+            if (requestHeadersParam.containsKey("Cookie") && requestHeadersParam.get("Cookie") instanceof String) {
+                String cookieString = (String) requestHeadersParam.get("Cookie");
+                if (remote != null) {
+                    // NOTE: In case that the path of URL is not end with `/`,
+                    //       last segment of a path is not stored as a path of Cookie.
+                    Cookie cookie = Cookie.parse(HttpUrl.get(remote), cookieString);
+                    if (cookie != null && replicationInternal != null) {
```

Comment:
Does it also need to remove the Cookies from the requestHeadersParams as well? I don't know the behavior when the cookies are in both headers and cookie jar.